pkg/query-service/app/logs/v3/query_builder.go

```diff
@@ -0,0 +1,349 @@
+	v3.AggregateOperatorP05:         0.05,
+	v3.AggregateOperatorP10:         0.10,
+	v3.AggregateOperatorP20:         0.20,
+	v3.AggregateOperatorP25:         0.25,
+	v3.AggregateOperatorP50:         0.50,
+	v3.AggregateOperatorP75:         0.75,
+	v3.AggregateOperatorP90:         0.90,
+	v3.AggregateOperatorP95:         0.95,
+	v3.AggregateOperatorP99:         0.99,
+	v3.AggregateOperatorHistQuant50: 0.50,
+	v3.AggregateOperatorHistQuant75: 0.75,
+	v3.AggregateOperatorHistQuant90: 0.90,
+	v3.AggregateOperatorHistQuant95: 0.95,
+	v3.AggregateOperatorHistQuant99: 0.99,
+}
+
+var aggregateOperatorToSQLFunc = map[v3.AggregateOperator]string{
+	v3.AggregateOperatorAvg:     "avg",
+	v3.AggregateOperatorMax:     "max",
+	v3.AggregateOperatorMin:     "min",
+	v3.AggregateOperatorSum:     "sum",
+	v3.AggregateOperatorRateSum: "sum",
+	v3.AggregateOperatorRateAvg: "avg",
+	v3.AggregateOperatorRateMax: "max",
+	v3.AggregateOperatorRateMin: "min",
+}
+
+var logsOperatorsEnabled = map[v3.FilterOperator]struct{}{
+	v3.FilterOperatorEqual:           {},
+	v3.FilterOperatorNotEqual:        {},
+	v3.FilterOperatorLessThan:        {},
+	v3.FilterOperatorLessThanOrEq:    {},
+	v3.FilterOperatorGreaterThan:     {},
+	v3.FilterOperatorGreaterThanOrEq: {},
+	v3.FilterOperatorLike:            {},
+	v3.FilterOperatorNotLike:         {},
+	v3.FilterOperatorRegex:           {},
+	v3.FilterOperatorNotRegex:        {},
+	v3.FilterOperatorIn:              {},
+	v3.FilterOperatorNotIn:           {},
+	// (todo) check contains/not contains/exists/not exists
+}
+
+// getClickhouseColumnName returns the corresponding clickhouse column name for the given attribute/resource key
+func getClickhouseColumnName(key v3.AttributeKey, fields map[string]v3.AttributeKey) (string, error) {
+	clickhouseColumn := key.Key
+	//if the key is present in the topLevelColumn then it will be only searched in those columns,
+	//regardless if it is indexed/present again in resource or column attribute
+	_, isTopLevelCol := constants.LogsTopLevelColumnsV3[key.Key]
+	if !isTopLevelCol && !key.IsColumn {
+		// check if we need to enrich metadata
+		if key.Type == "" || key.DataType == "" {
+			// check if the field is present in the fields map
+			if field, ok := fields[key.Key]; ok {
+				if field.IsColumn {
+					return field.Key, nil
+				}
+				key.Type = field.Type
+				key.DataType = field.DataType
+			} else {
+				return "", fmt.Errorf("field not found to enrich metadata")
+			}
+		}
+
+		columnType := key.Type
+		if columnType == v3.AttributeKeyTypeTag {
+			columnType = "attributes"
+		} else {
+			columnType = "resources"
+		}
+
+		columnDataType := "string"
+		if key.DataType == v3.AttributeKeyDataTypeFloat64 {
+			columnDataType = "float64"
+		}
+		if key.DataType == v3.AttributeKeyDataTypeInt64 {
+			columnDataType = "int64"
+		}
+		clickhouseColumn = fmt.Sprintf("%s_%s_value[indexOf(%s_%s_key, '%s')]", columnType, columnDataType, columnType, columnDataType, key.Key)
```

Comment:
Are booleans skipped somewhere? Otherwise, they would result in the wrong query for the query range?

Reply:
Right, as of now in the OTEL collector bool data is stored as a string https://github.com/SigNoz/signoz-otel-collector/blob/7338768fa6b2c61aa8c46f45c9c7de700e2795b8/exporter/clickhouselogsexporter/migrations/000001_init_db.up.sql#L14 

Have added comments in the code for that.